### PR TITLE
Collected small changes and fixes

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

Various small changes and fixes

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

The following individual changes are included:
- make `Output::add_dump()` return a pointer to the created dump
- make "write_dump" use `Output::add_dump()` this avoids segfaults with some dump_modify options
- fix "write_dump" internal dump style creation every option to avoid bogus "fix not computed at compatible times" errors
- fix out-of-bounds access bug with custom keywords for dump style cfg (detected by MSVC only!)
- refactor processing of arguments for hybrid styles to be similar to pair style hybrid
- arguments to bond/angle/dihedral/improper styles that don't use arguments are no longer silently ignored
- fix some minor documentation issues
- add "creator:" tag to YAML dump style
- correctly handle the case that the PyYAML module is not installed in the LAMMPS python module and the unit tests
- request that libyaml-cpp is always build as a static library so it won't need to be installed whan LAMMPS is installed
- request that the MDI library is always built with position independent code
- apply clang-tidy fixes to recently modified code and files that have been previously not included in "tidying"
- update singularity definition files for CUDA 11.6 and ROCm 5.1.2

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

File: src/DPD-REACT/random_external_state.h [hunk 1]
```diff
@@ -111,7 +111,7 @@ int es_rand(es_RNG_t &state_)
 LAMMPS_INLINE
 double es_drand(es_RNG_t &state_)
 {
-  return 1.0 * es_urand64(state_) / MAX_URAND64;
+  return static_cast<double>(es_urand64(state_) / MAX_URAND64);
```
Comment (akohlmey) [inline src/DPD-REACT/random_external_state.h]: are you sure this should not be?
```
static_cast<double>(es_urand64(state_)) / MAX_URAND64;
```

Comment (stanmoore1) [reply src/DPD-REACT/random_external_state.h]: @akohlmey Good point, I tweaked the change a little, trying to fix this warning:

```
../random_external_state.h:114:52: warning: implicit conversion from 'const uint64_t' (aka 'const unsigned long') to 'double' changes value from 18446744073709551614 to 18446744073709551616 [-Wimplicit-const-int-float-conversion]
  return static_cast<double>(es_urand64(state_)) / MAX_URAND64;
                                                 ~ ^~~~~~~~~~~
  ```

Comment (stanmoore1) [reply src/DPD-REACT/random_external_state.h]: Done--thanks

**Review by stanmoore1 — APPROVED** [review]
I approve

**Review by sjplimp — APPROVED** [review]

**Pull request merged**